### ISSUE TITLE
Parallelize push and pull

### ISSUE THREAD

**Issue by ffittschen** [phrase/phrase-cli#40]
As far as I can see, `push` and `pull` currently [loop through](https://github.com/phrase/phraseapp-client/blob/master/push.go#L145) the `localeFiles` and upload each individual file after another. For us this takes roughly ~40 seconds depending on the internet connection. If we also pass the `--wait` option this issue is greatly amplified and it takes ~15 minutes! 

Could this be parallelized in a way to distribute the uploads / downloads across multiple threads/connections to speed this up a bit? Especially for the uploads it would be helpful to continue other uploads while waiting, instead of doing everything sequentially.

**Comment by egze** [comment]
We have decided to close all GitHub issues and, instead, handle them through our Customer Support. We think this shift will allow us to offer proper prioritization of issues as well as a more appropriate response time that would be expected from Phrase products.

We apologize for any inconveniences this may cause. If you believe your issue is still relevant or has a high priority, please contact support@phrase.com, and we can take more direct action toward finding a solution.

**Comment by tetienne** [comment]
Almost 3 years later, there is still no such feature. To speed up the process and avoid multiple the cli could download an archive and continue the process as usual. 

@egze Please consider re opening this issue.

**Comment by theSoenke** [comment]
Hey @tetienne, right now we don't have immediate plans to upload and download locales files in parallel. The API also has a concurrency limit of 4, which will limit the possible speed-up gains. If you're using the `--wait` flag I would recommend to remove it unless you absolutely need to wait until each upload has been processed.

But we're also considering looking into client side caching where it doesn't have to pull the same content again when it didn't change. That should also make the pull command a lot faster in many cases

**Comment by tetienne** [comment]
@theSoenke thx for taking care of my comment.
Sadly, it won’t help with our case, we are using `phrase pull` within our Dockerfile. So when the cache is dirty, most of the build time is wasted by phrase. It can last around 40~50s.

**Comment by tetienne** [comment]
> But we're also considering looking into client side caching where it doesn't have to pull the same content again when it didn't change. That should also make the pull command a lot faster in many cases

@theSoenke Any update on this

**Comment by theSoenke** [comment]
@tetienne sorry, there's no update yet. Client side caching likely also wouldn't help much when used inside of docker. Without a persistent storage the cache would always be empty. I don't know whether an option for you, but one solution could be to regularly update translations with the GitHub action: https://github.com/phrase/setup-cli For example on merge into a specific branch or in regular intervals
